### PR TITLE
Bugfix/dogfooding fixes

### PR DESCRIPTION
**Pull request by PauloSousapt**
This PR fixes partially #210 and #243 

## Tests

- [x] Tested on mobile
- [x] Tested on dektop

This PR should be manually tested by @hugomarquesdev , included on reviewers.

PS: This PR needs to be merged after the backend state update in Beneficiares/Managers

### CODE REVIEW

File: src/components/Table.tsx [hunk 1]
```diff
@@ -5,12 +5,15 @@ import React, { useEffect, useRef, useState } from 'react';
...
     [P in keyof BaseTableProps]?: BaseTableProps[P];
 };
 
 type TableProps = {
-    callback: Function;
+    prefix: string;
+    callback?: Function;
     callbackProps?: Object;
     itemsPerPage: number;
     refresh?: Date;
```
Comment (hugomarquesdev) [inline src/components/Table.tsx:19]: sort

File: src/views/Beneficiaries/index.tsx [hunk 2]
```diff
@@ -40,38 +37,27 @@ const Beneficiaries: React.FC<{ isLoading?: boolean }> = props => {
...
+        if(!getByKey('state')) {
+            router.push('/manager/beneficiaries?state=0&orderBy=since%3Adesc', undefined, { shallow: true });
+        }
     }, []);
 
-    const onAddBeneficiary = () => {
-        setRefresh(new Date());
-        communityBeneficiaries.refetch();
-    }
+    console.log(data?.beneficiaryEntities?.length);
```
Comment (hugomarquesdev) [inline src/views/Beneficiaries/index.tsx:57]: 👀

**Review by hugomarquesdev — CHANGES_REQUESTED** [review]

File: src/components/Table.tsx [hunk 3]
```diff
@@ -5,12 +5,15 @@ import React, { useEffect, useRef, useState } from 'react';
...
+import useBeneficiaries from '../hooks/useBeneficiaries';
+
 type Partial<BaseTableProps> = {
     [P in keyof BaseTableProps]?: BaseTableProps[P];
 };
 
 type TableProps = {
-    callback: Function;
+    prefix: string;
+    callback?: Function;
```
Comment (psoaresbj) [inline src/components/Table.tsx:16]: sort

File: src/components/Table.tsx [hunk 4]
```diff
@@ -24,107 +27,52 @@ const sortToString = (sort: any) => {
...
-            sortDesc: parts[1] === 'desc'
-        };
-    }
-
-    return {};
-}
-
 const Table: React.FC<TableProps & Partial<BaseTableProps>> = props => {
-    const { columns, callback, callbackProps, itemsPerPage, refresh, ...forwardProps } = props; 
+    const { columns, callback, callbackProps, itemsPerPage, refresh, prefix, ...forwardProps } = props;
```
Comment (psoaresbj) [inline src/components/Table.tsx:31]: sort

File: src/hooks/useBeneficiaries.ts [hunk 5]
```diff
@@ -0,0 +1,42 @@
+import { selectCurrentUser } from '../state/slices/auth';
+import { useSelector } from 'react-redux';
+import config from '../../config';
+import useSWR from 'swr';
+
```
Comment (psoaresbj) [inline src/hooks/useBeneficiaries.ts:5]: remove ln

File: src/hooks/useBeneficiaries.ts [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+    const queryString = Object.keys(incomingFilters)
+        .map((key) => `${ key }=${ incomingFilters[key] }`)
+        .join('&');
+
+    const fetcher = (url: string) => fetch(config.baseApiUrl + url, {
+        headers: { Authorization: `Bearer ${auth.token}` }
+    }).then((res) => res.json());
+
+    const { data, mutate, error } = useSWR(`/communities/${entity}?${queryString}`, fetcher);
+
```
Comment (psoaresbj) [inline src/hooks/useBeneficiaries.ts:28]: remove ln

File: src/modals/Contribute.tsx [hunk 7]
```diff
@@ -188,6 +188,7 @@ const Contribute = () => {
                         </Box>
                     </Box>
                     <Input
+                        // type="numeric"   ADD support for numeric keyboard on UI
```
Comment (psoaresbj) [inline src/modals/Contribute.tsx:191]: dead code?

File: src/views/Communities/Community.tsx [hunk 8]
```diff
@@ -61,7 +60,7 @@ const Communities = ({ community }: any) => (
                             regular
                             small
                         >
-                            {community.city}
+                            { community.city.length > 19 ? `${community.city.substring(0, 19)}...` : community.city }
```
Comment (psoaresbj) [inline src/views/Communities/Community.tsx:63]: trim white spaces

**Review by psoaresbj — APPROVED** [review]

**Review by hugomarquesdev — APPROVED** [review]

**Pull request merged**